client/src/containers/PassResetContainer.tsx

```diff
@@ -0,0 +1,251 @@
+  };
+  
+  const validateConfirmPass = (): boolean => {
+    if (passResetForm.password !== confirmPass) {
+      const errMsg = "Passwords must be matching";
+      setConfirmPassError(errMsg);
+      return false;
+    } else {
+      setConfirmPassError("");
+      return true;
+    }
+  };
+
+  const handleFormInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPassResetForm({ ...passResetForm, [e.target.name]: e.target.value });
+  };
+
+  const handleError =
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>) =>
+    (msg: string) => {
+      return setPassResetFormErrors({
+        ...passResetFormErrors,
+        [e.target.name]: msg,
+      });
+    };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const msg = getErrorMessages();
+    if (msg !== "") {
+      console.warn(msg);
+      showAlert("warning", msg);
+      return;
+    }
+
+    if (isFormMissingFields()) {
+      const msg = "Please fill in all the fields.";
+      console.warn(msg);
+      showAlert("warning", msg);
+      return;
+    }
+
+    try {
+      const { status, data } = await api.resetPass({...passResetForm, token: token });
+      if (status !== 204) {
+        if (!data) {
+          throw new Error("Missing error response.");
+        }
+        const msg = data.message;
+        console.error(msg);
+        showAlert("error", msg);
+      } else {
+        showAlert("success", "Password has been reset! You may now close this tab.");
+        // navigate to a new page that displays this message instead of using alert
+      }
+    } catch (error) {
+      console.error(error);
+      showAlert(
+        "error",
+        "An error occurred while resetting your password. Please try again later."
+      );
+    }
+  };
+
+  return (
+    <Container maxWidth='xs'>
+      <Box
+        sx={{
+          marginTop: 8,
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+        }}
+      >
+        <Snackbar open={alert.display} onClose={hideAlert}>
+          <Alert
+            onClose={hideAlert}
+            severity={alert.severity as AlertColor}
+          >
+            <Typography sx={{ whiteSpace: "pre-line" }}>
+              {alert.message}
+            </Typography>
+          </Alert>
+        </Snackbar>
+        <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
+          <LockOutlinedIcon />
+        </Avatar>
+        <Typography component='h1' variant='h5'>
+          Create new password
+        </Typography>
+        <Box component='form' onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
+          <TextField
+            margin='normal'
+            required
+            fullWidth
+            name='password'
+            label='New Password'
+            type='password'
+            data-testid='password'
+            onChange={handleFormInput}
+            onBlur={(e) =>
+              validateBlur(
```

Comment:
add a RNG that has 1/100 chance to say "New password cannot be old password"

Reply:
lol